packages/web/components/trade-clipboard/index.tsx

```diff
@@ -0,0 +1,481 @@
+                  <li
+                    className={classNames(
+                      "flex items-center justify-center w-full h-8 cursor-pointer rounded-full",
+                      slippageConfig.isManualSlippage
+                        ? "text-white-high"
+                        : "text-white-faint",
+                      slippageConfig.isManualSlippage
+                        ? slippageConfig.getManualSlippageError()
+                          ? "bg-missionError"
+                          : "bg-primary-200"
+                        : "bg-background"
+                    )}
+                    onClick={(e) => {
+                      e.preventDefault();
+
+                      if (manualSlippageInputRef.current) {
+                        manualSlippageInputRef.current.focus();
+                      }
+                    }}
+                  >
+                    <AutosizeInput
+                      inputRef={(ref) => {
+                        manualSlippageInputRef.current = ref;
+                      }}
+                      inputClassName="bg-transparent text-center"
+                      minWidth={0}
+                      value={slippageConfig.manualSlippageStr}
+                      onChange={(e) => {
+                        e.preventDefault();
+
+                        slippageConfig.setManualSlippage(e.target.value);
+                      }}
+                      onFocus={(e) => {
+                        e.preventDefault();
+
+                        slippageConfig.setIsManualSlippage(true);
+                      }}
+                    />
+                    <span className="shrink-0">%</span>
+                  </li>
+                </ul>
+              </div>
+            )}
+          </div>
+
+          <div className="relative">
+            <div className="bg-surface rounded-2xl px-4 pt-3 pb-4 relative">
+              <div className="flex justify-between items-center">
+                <span className="subtitle1 text-white-full">From</span>
+                <div className="flex items-center">
+                  <span className="caption text-sm text-white-full">
+                    Available
+                  </span>
+                  <span className="caption text-sm text-primary-50 ml-1.5">
+                    {availableBalance
+                      .trim(true)
+                      .shrink(true)
+                      .maxDecimals(6)
+                      .toString()}
+                  </span>
+                  <button
+                    type="button"
+                    className={classNames(
+                      "text-white-full text-xs py-1 px-1.5 rounded-md ml-2",
+                      tradeTokenInConfig.fraction === 1
+                        ? "bg-primary-200"
+                        : "bg-white-faint"
+                    )}
+                    onClick={(e) => {
+                      e.preventDefault();
+
+                      if (tradeTokenInConfig.fraction !== 1) {
```

Comment:
Seems like common functionality, we could have a `toggleFraction()` method on the config object?

Reply:
`fraction` accepts any value between 0 and 1. This code itself isn't much repeated imo, I think we could leave it like this.